corehq/apps/hqadmin/views.py

```diff
@@ -490,11 +492,18 @@ def get_context_data(self, **kwargs):
         context = super(AdminRestoreView, self).get_context_data(**kwargs)
         response, timing_context = self._get_restore_response()
         timing_context = timing_context or TimingContext(self.user.username)
-        string_payload = ''.join(response.streaming_content)
-        xml_payload = etree.fromstring(string_payload)
-        restore_id_element = xml_payload.find('{{{0}}}Sync/{{{0}}}restore_id'.format(SYNC_XMLNS))
-        num_cases = len(xml_payload.findall('{http://commcarehq.org/case/transaction/v2}case'))
-        formatted_payload = etree.tostring(xml_payload, pretty_print=True)
+        if isinstance(response, StreamingHttpResponse):
+            string_payload = ''.join(response.streaming_content)
+            xml_payload = etree.fromstring(string_payload)
+            restore_id_element = xml_payload.find('{{{0}}}Sync/{{{0}}}restore_id'.format(SYNC_XMLNS))
+            num_cases = len(xml_payload.findall('{http://commcarehq.org/case/transaction/v2}case'))
+            formatted_payload = etree.tostring(xml_payload, pretty_print=True)
+        else:
+            # response is HttpResponse 500
```

Comment:
we don't actually know this to be true right - just that it's an `HttpResponse`? could add a check for the status code or just update the comment to say something like "response is HttpResponse which is unexpected - tell the user something might have gone wrong".

Then I think we should also add a `messages.error` message along the lines of: "unexpected restore response - if you believe this is a bug please report an issue" so it's clear something has gone wrong to the user and they can do something about it.